src/Handler.ts

```diff
@@ -339,20 +347,77 @@ class Handler extends Eventful {
         const out = new HoveredResult(x, y);
 
         for (let i = list.length - 1; i >= 0; i--) {
+            const el = list[i];
             let hoverCheckResult;
-            if (list[i] !== exclude
+            if (el !== exclude
                 // getDisplayList may include ignored item in VML mode
-                && !list[i].ignore
-                && (hoverCheckResult = isHover(list[i], x, y))
+                && !el.ignore
+                && (hoverCheckResult = isHover(el, x, y))
             ) {
-                !out.topTarget && (out.topTarget = list[i]);
+                !out.topTarget && (out.topTarget = el);
                 if (hoverCheckResult !== SILENT) {
-                    out.target = list[i];
+                    out.target = el;
                     break;
                 }
             }
         }
 
+        if (this._pointerSize && !out.target) {
+            /**
+             * If no element at pointer position, check intersection with
+             * elements with pointer enlarged by target size.
+             */
+            const candidates = [];
+            const pointerSize = this._pointerSize;
+            const targetSizeHalf = pointerSize / 2;
+            const pointerRect = new BoundingRect(x - targetSizeHalf, y - targetSizeHalf, pointerSize, pointerSize);
+
+            for (let i = list.length - 1; i >= 0; i--) {
+                const el = list[i];
+                if (el !== exclude
+                    && !el.ignore
+                    && !el.ignoreCoarsePointer
+                    // If an element ignores, its textContent should also ignore.
+                    // TSpan's parent is not a Group but a ZRText.
+                    // See Text.js _getOrCreateChild
+                    && (!el.parent || !(el.parent as any).ignoreCoarsePointer)
+                ) {
+                    tmpRect.copy(el.getBoundingRect());
+                    if (el.transform) {
+                        tmpRect.applyTransform(el.transform);
+                    }
+                    if (tmpRect.intersect(pointerRect)) {
+                        candidates.push(el);
+                    }
+                }
+            }
+
+            /**
+             * If there are elements whose bounding boxes are near the pointer,
+             * use the most top one that intersects with the enlarged pointer.
+             */
+            if (candidates.length) {
+                const rStep = 4;
+                const thetaStep = Math.PI / 12;
+                for (let r = 0; r < targetSizeHalf && !out.target; r += rStep) {
+                    for (let theta = 0; theta < Math.PI * 2 && !out.target; theta += thetaStep) {
+                        const x1 = x + r * Math.cos(theta);
+                        const y1 = y + r * Math.sin(theta);
+                        for (let i = 0; i < candidates.length; i++) {
```

Comment:
This logic is same to https://github.com/ecomfe/zrender/blob/85d6614900e127c3abcc81fef58f496e14fe7eba/src/Handler.ts#L375 . Perhaps we can extract it to a separate function to avoid writing this complex `target`, `topTarget` and `SILENT` things twice.

And don't forget to reverse the iteration order in https://github.com/ecomfe/zrender/pull/918/files#diff-2befe9e583e17eed633cc9bd738c9b0c54ffb1f48222871eda2cc4dc232c48eeR375 after this change